packages/react/src/components/Menu/MenuItem.tsx

```diff
@@ -362,12 +372,25 @@ class MenuItem extends AutoControlledComponent<ReactProps<MenuItemProps>, MenuIt
     const { menu } = this.props
     const { menuOpen } = this.state
     if (menu && !menuOpen) {
-      this.setState({ menuOpen: true })
+      this.trySetMenuOpen(true, e)
       _.invoke(this.props, 'onActiveChanged', e, { ...this.props, active: true })
       e.stopPropagation()
       e.preventDefault()
     }
   }
+
+  private trySetMenuOpen(newValue: boolean, eventArgs: any, onStateChanged?: any) {
```

Comment:
```suggestion
  private trySetMenuOpen(e: React.SyntethicEvent, newValue: boolean, onStateChanged?: any) {
```

nit: Can we place `e` param on the first place as in other methods?

Reply:
fyi, the order is the same as in Popup for trySetOpen method. But I can change it in MenuItem to keep consistency throughout the component.